src/components/element-editor/element-editor.tsx

```diff
@@ -0,0 +1,124 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Button, Form } from "@lifesg/react-design-system";
+import { useMemo } from "react";
+import { FormProvider, useForm, useFormContext } from "react-hook-form";
+import { useBuilder } from "src/context-providers";
+import * as yup from "yup";
+import { IFieldValues } from "./types";
+
+export const ElementEditor = () => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const { elements, focusedElement } = useBuilder();
+
+    const camelCaseRegex = /^[a-z]+((\d)|([A-Z0-9][a-z0-9]+))*([A-Z])?$/gm;
+    const schema = useMemo(() => {
+        return yup.object<IFieldValues>().shape({
```

Comment:
The changes we talked about in the CR is not reflected here. We mentioned to move the schemas into another file instead of adding it in the editor.